lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -187,14 +183,14 @@ def test_bad_dates(self, client, server_config, user_ok, pbench_token):
         """
         Test behavior when a bad date string is given
         """
-        if all(
-            p.type != ParamType.DATE for _, p in self.cls_obj.schema.parameters.items()
-        ) or any(k not in self.payload for k in ("start", "end")):
+        skip = True
+        for key, p in self.cls_obj.schema.parameters.items():
+            # Modify date/time key in the payload to make it look invalid
+            if p.type == ParamType.DATE and key in self.payload:
+                self.payload[key] = "2020-19"
+                skip = False
+        if skip:
             pytest.skip("skipping " + self.test_bad_dates.__name__)
```

Comment:
You shouldn't need the `skip` flag -- I believe the following will accomplish the same thing:
```
        for key, p in parameter_items:
            # Modify date/time key in the payload to make it look invalid
            if p.type == ParamType.DATE and key in self.payload:
                self.payload[key] = "2020-19"
                break
        else:
            pytest.skip("skipping " + self.test_bad_dates.__name__)
```
So, once a `DATE` field is found, it'll break out of the loop; only if it fails to break out does it execute the `else` clause.  (Note that I also replaced your use of `self.cls_obj.schema.parameters.items()` with the local `parameter_items` variable.)

However, I don't think that this will accomplish what Dave requested:  this is going to replace only the first `DATE` field that it encounters with a bad date, and then it's going to make a request with the resulting payload, and then the test will end -- so it's really only going to test _one_ of the date fields (i.e., if there is more than one).

Reply:
Technically, yes that should work since our code will abort with `BAD_REQUEST` as soon as it encounters first bad date/time field so that means we only need one bad date/time field in the payload for the test to succeed.

Reply:
That depends greatly on what you mean by "work" and "succeed"!  😁 

If our Server code handles bad data in `"start"` gracefully but crashes on bad data in `"end"`, then we won't know about that if our unit test only tries bad data in the `"start"` field.

Part of the idea behind unit tests is to try to get execution to follow all the paths through the CUT.  It's tricky to do this without making the unit tests know too much about the CUT, but, at a minimum, it's reasonable to test all the parameters to our interface in as many combinations as we can easily generate.

Thus, testing only the first one is not very satisfying, especially if we can put it in a loop which tests each of them.

Reply:
> If our Server code handles bad data in `"start"` gracefully but crashes on bad data in `"end"`

I highly doubt this will happen, because all the date/time fields we parse via parser from dateutil, it wont handle one field gracefully and crash on another same type of field. 

But I guess if we want to make sure that all the date/time field gets the bad date then this code wont achieve that. Alternately, I can move the code that makes the post request (with bad date payload) inside this for loop and we can test for each date/time field present in the payload for the bad date, if that make sense. @webbnh @dbutenhof